### DISCUSSION TITLE
hide/show Minimap

### DISCUSSION THREAD

**Discussion by itsmesoh** [xyflow/xyflow#1950] [Q&A] [answered]
I would like to have the option to hide /show minimap. I guess there is no property for this. 
So was trying to create a new button using Extended controls and onclick call the minimap.

It doesnt seem to work. Can someone help here?

Thanks

**Comment by moklick** [comment, marked as answer]
You can use a flag for that. Something like this should work:

```jsx
function Flow() {
  const [isMinimapVisible, setIsMinimapVisible] = useState(false);
  const onToggleMinimap = useCallback(() => {
    setIsMinimapVisible(isVisible => !isVisible);
  }, []);
 
  return (
    <ReactFlow
      elements={[]}
    >
     {isMinimapVisible && <MiniMap />}
     <button
      onClick={onToggleMinimap}
      style={{ position: 'absolute', zIndex: 1000 }}
    >
      toggle minimap</button>
    </ReactFlow>
  );
}

```

**Reply by itsmesoh** [reply]
This works.

Thank you for the quick response. Much appreciated :)
